### DISCUSSION TITLE
API parameters cause 404 error

### DISCUSSION THREAD

**Discussion by elavy-harris** [nextgenhealthcare/connect#6141] [Q&A] [answered]
Mirth 4.5.0

I get messages back when I try

`curl -X GET -su elavy -k "https://interface.harris-chr.com:8443/api/channels/5c19b0fe-251b-40ce-a376-51e5951ba96e/messages" -H X-Requested-With:whatever -H accept:application/json`

I get a 404 error, however, when I try

`curl -X GET -su elavy -k "https://interface.harris-chr.com:8443/api/channels/5c19b0fe-251b-40ce-a376-51e5951ba96e/messages?startDate=2020-01-01T00%3A00%3A00Z&endDate=2025-01-01T00%3A00%3A00Z" -H X-Requested-With:whatever -H accept:application/json`

The only difference between the two is the addition of `?startDate=x&endDate=y`. (Using colons instead of %3A doesn't change anything.)

Any ideas?

**Comment by pacmano1** [comment]
https://github.com/nextgenhealthcare/connect/discussions/4638#discussioncomment-2090314

Note in the comments in the thread it links to another issue, so maybe fixed in later releases. 

And I know you know better to post your mirth version :)

**Reply by elavy-harris** [reply]
Sorry. I've now added a reference to Mirth 4.5.0 to the description.

**Reply by elavy-harris** [reply]
No luck. Still got a 404 with

`curl -X GET -su elavy -k "https://interface.harris-chr.com:8443/api/channels/5c19b0fe-251b-40ce-a376-51e5951ba96e/messages?startDate=2020-01-01T00%3A00%3A00Z&endDate=2025-01-01T00%3A00%3A00Z&limit=1&offset=0&includeContent=false" -H X-Requested-With:whatever -H accept:application/json`

**Comment by pacmano1** [comment]
```shell
curl -k -u admin:somepassword  -X GET "https://192.168.66.104:8443/api/channels/3185d5f6-08fa-4f63-af3c-013ffbea4b5e/messages?startDate=2024-03-01T09%3A00%3A00.000-0700&endDate=2024-03-29T09%3A00%3A00.000-0700&status=SENT&includeContent=false&offset=0&limit=20" -H "accept: application/json" -H "X-Requested-With: OpenAPI"
```

Works fine for me.

**Comment by elavy-harris** [comment, marked as answer]
Thanks @pacmano1 for getting me going in the right direction. This works:

`curl -X GET -su elavy -k "https://interface.harris-chr.com:8443/api/channels/5c19b0fe-251b-40ce-a376-51e5951ba96e/messages?startDate=2020-01-01T00%3A00%3A00.000-0000&endDate=2025-01-01T00%3A00%3A00.000-0000" -H X-Requested-With:whatever -H accept:application/json`

It appears that (a) the milliseconds must be included and (b) the timezone must be specified as an offset. For positive offsets, the plus sign must be specified as %3B.

**Comment by elavy-harris** [comment]
Opened issue #6144
